### PR TITLE
feat(serialize/payload): improve ser/de and control payload

### PR DESCRIPTION
**Pull request by joshuachp**
Use Serialize and Deserialize to not allocate an additional HashMap when working with the payload. Separate the types errors int the specific module. Check the length of the control payload after deflating it and remove the unwraps.

### CODE REVIEW

File: src/payload.rs [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+            Ok(Aggregation::Object(hmap))
+        }
+        value => {
+            let individual = AstarteType::try_from(value)?;
+
+            Ok(Aggregation::Individual(individual))
+        }
+    }
+}
```
Comment (harlem88) [inline src/payload.rs:124]: Move this function below `fn serialize_object`.

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/payload.rs
Comment (rgallor) [inline src/payload.rs:1]: Some pub functions/methods are missing doc comments.

File: src/properties.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    let (size, data) = bdata.split_at(4);
+    // The size is a u32 in big endian, so we need to convert it to usize
+    let size: u32 = u32::from_be_bytes([size[0], size[1], size[2], size[3]]);
+    let size: usize = size.try_into()?;
+
+    let mut d = ZlibDecoder::new(data);
+    let mut s = String::new();
+    let bytes_red = d.read_to_string(&mut s)?;
```
Comment (rgallor) [inline src/properties.rs:56]: should be `bytes_read` (not `red`).

**Review by rgallor — CHANGES_REQUESTED** [review]

**Review by sorru94 — APPROVED** [review]

File: src/payload.rs (hunk 1 above)
Comment (joshuachp) [reply src/payload.rs:124]: Done

File: src/properties.rs (hunk 2 above)
Comment (joshuachp) [reply src/properties.rs:56]: Fixed

**Review by rgallor — APPROVED** [review]

**Pull request merged**